### PR TITLE
log Zyte API request id on errors

### PR DESCRIPTION
**Pull request by BurnzZ**
TODO: 
- [x] Use new `python-zyte` version containing:
    - https://github.com/zytedata/python-zyte-api/pull/52
    - https://github.com/zytedata/python-zyte-api/pull/54

### CODE REVIEW

File: scrapy_zyte_api/handler.py [hunk 1]
```diff
@@ -218,10 +218,12 @@ async def _download_request(
                 retrying=retrying,
             )
         except RequestError as er:
+            request_id = er.headers.get("request-id")
             error_detail = (er.parsed.data or {}).get("detail", er.message)
```
Comment (kmike) [inline scrapy_zyte_api/handler.py:222]: Should we include request-id to the python-zyte-api error message instead?

**Review by lopuhin — APPROVED** [review]
Looks great, thanks @BurnzZ 👍

File: scrapy_zyte_api/handler.py (hunk 1 above)
Comment (BurnzZ) [reply scrapy_zyte_api/handler.py:222]: Something like this? https://github.com/zytedata/python-zyte-api/pull/52

**Review by wRAR — APPROVED** [review]

File: scrapy_zyte_api/handler.py [hunk 2]
```diff
@@ -220,8 +220,9 @@ async def _download_request(
         except RequestError as er:
             error_detail = (er.parsed.data or {}).get("detail", er.message)
             logger.error(
-                f"Got Zyte API error (status={er.status}, type={er.parsed.type!r}) "
-                f"while processing URL ({request.url}): {error_detail}"
+                f"Got Zyte API error (status={er.status}, type={er.parsed.type!r}, "
+                f"request_id='{er.request_id}') while processing URL ({request.url}): "
```
Comment (Gallaecio) [inline scrapy_zyte_api/handler.py:224]:
```suggestion
                f"request_id={er.request_id!r}) while processing URL ({request.url}): "
```

**Review by Gallaecio — APPROVED** [review]

**Pull request merged**